packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -309,11 +411,22 @@ export class AzureCommunicationCallAdapter implements CallAdapter {
       return backendId;
     });
 
-    return this.handlers.onStartCall(idsToAdd);
+    try {
+      return this.handlers.onStartCall(idsToAdd);
+    } catch (e) {
+      this.context.setError(e);
+      this.emitter.emit('error', e);
+      return undefined;
+    }
   }
 
   public async removeParticipant(userId: string): Promise<void> {
-    this.handlers.onParticipantRemove(userId);
+    try {
+      await this.handlers.onParticipantRemove(userId);
+    } catch (e) {
+      this.context.setError(e);
+      this.emitter.emit('error', e);
+    }
   }
```

Comment:
Should we extract a generic error handling method to DRY things up? It seems the pattern is, we want to handle errors by setting it to the context and then emitting an event. In some cases, we want to return a default value after that. Thinking something like this:

```TypeScript
withErrorHandling(async () => {
  await this.handlers.onParticipantRemove(userId);
})

// When we have a default value

withErrorHandling(async () => {
  return this.deviceManager.getMicrophones();
},  Promise.resolve([] as AudioDeviceInfo[]))
```

Reply:
Thanks Sohan, I think that's a good idea, I've updated this PR with the withErrorHandling wrapper function.